client/src/App.tsx

```diff
@@ -19,28 +21,30 @@ function App(): JSX.Element {
         <SnackBarProvider>
           <AuthProvider>
             <SocketProvider>
-              <NavBar />
-              <Switch>
-                <Route exact path="/login" component={Login} />
-                <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/my-jobs">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/messages">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/my-sitters">
-                  <Dashboard />
-                </Route>
-                <Route exact path="/edit-profile" component={EditMenu} />
+              <RequestProvider>
```

Comment:
Do we want to wrap the entire app with the `RequestProvider`? Each time we receive a request the app will be re-rendered. Take for instance, we are editing our profile and a request has been received - this will re-render the entire app, impacting overall app performance :)

Reply:
I moved it to wrap the sitters and job pages.